File: src/main/js/FindNthCharInFile.js
let fs = require('fs');
let q = require('q');

class FindNthCharInFile {
	static getEveryNthChar(pathToFile, n) {
		return FindNthCharInFile.getFileContents(pathToFile).then(function(fileContents) {
					let result = '';
					for (let i = 0; i < fileContents.length; i += n) {
						result += fileContents.charAt(i);
					}
					return result;
				}).catch(function(error) {
			console.log(error);
		});
	}

	static getFileContents(pathToFile) {
		let defer = q.defer();
		fs.readFile(pathToFile, 'utf8', function(error, data) {
			defer.resolve(data);
		});
		return defer.promise;
	}
}

module.exports = FindNthCharInFile;

File: src/main/java/FindNthCharInFile.java
import java.io.*;
import java.nio.ByteBuffer;
import java.nio.channels.FileChannel;
import java.nio.file.Files;
import java.nio.file.Paths;

public class FindNthCharInFile {
	public static String getEveryNthChar(String pathToFile, int n) throws IOException {
		String fileContents = getFileContents(pathToFile);

		StringBuilder result = new StringBuilder();
		for (int i = 0; i < fileContents.length(); i+=n) {
			result.append(fileContents.charAt(i));
		}

		return result.toString();
	}

	private static String getFileContents(String pathToFile) throws IOException {
		byte[] bytes = Files.readAllBytes(Paths.get(pathToFile));

		return new String(bytes, "US-ASCII");
	}
}
